### ISSUE TITLE
Candy Guard Mint Deprecated as of 1.0.0. Use MintV2 instead

### ISSUE THREAD

**Issue by Tristyn** [metaplex-foundation/js#493]
I have a Candy Machine UI, it fails to mint. The problem is `metaplex-foundation/js` uses the deprecated `createMintInstruction` from `@metaplex-foundation/mpl-candy-guard@0.3.0`, and `createMintV2Instruction` was introduced 5 days ago in `@0.4.1` 

This is really difficult because you already upgraded the program to deprecate V1 but haven't updated the library to support V2

A failed mint transaction: https://explorer.solana.com/tx/3Eg5QDfnaZuhHeqbrnhXms4LFRSKbmumzTgpvH6cBNJN7A1P1cnAaULYmX7BTcu4ZgbHpzxoMXZsZaSxWZVvYMvE
The relevant parts:
```
Candy Guard Instruction
> Program logged: "Instruction: Mint"
> Program logged: "(Deprecated as of 1.0.0) Use MintV2 instead"
> Program invoked: System Program
  > Program returned success
> Program invoked: System Program
  > Program returned success
> Program invoked: Unknown Program (CndyV3LdqHUfDLmE5naZjVN8rBZz4tqhdefbAnjHG3JR)
  > Program logged: "Instruction: MintV2"
...
  > Program logged: "AnchorError occurred. Error Code: MissingTokenAccount. Error Number: 6025. Error Message: Missing token account."
  > Program consumed: 131061 of 913768 compute units
  > Program returned error: "custom program error: 0x1789"
> Program consumed: 189533 of 972240 compute units
> Program returned error: "custom program error: 0x1789"
```

**Comment by lorisleiva** [comment]
Hi there,

Mint v1 should still work exactly the same way it was before even after the program upgrades. The deprecation message is just a warning to let people know about the latest mint instruction.

Reading your error message, it looks like the token account is missing from the mint instruction.

**Comment by Tristyn** [comment]
Thank you so much for the response

[I attempted to provide a token account but it did not work.](https://github.com/joinpoolparty/candy-machine-v3-ui/commit/24712ae6fe1b62777583cd95841b566a407c868c)

Is there any candy machine v3 ui today that works? Reskinning a UI is the easiest way to get started. I've tried
https://github.com/wearekite/candy-machine-v3-ui
https://github.com/Solana-Studio/Candy-Machine-V3-UI

If not, I will have to dig through v3 docs, but I have a feeling there is a bug on the loose

**Comment by Tristyn** [comment]
Unfortunately this means Sugar CLI doesn't have a use until the new sdk rolls out, I like Sugar a lot :(

**Comment by lorisleiva** [comment]
Hey, thanks for adding more info.

We're working on adding a more helpful error message on the candy guard program when a mint instruction is used with a candy machine v2.

Regarding Sugar, you can use a previous version (I think 2.0) to create candy machines v1 which will be compatible with the JS SDK. We'll add a warning on the documentation as well.

**Comment by Iceee1000** [comment]
I'm having this same issue but am more confused after reading this. Docs say the latest sugar cli makes a v3 candy machine (https://docs.metaplex.com/programs/candy-machine/how-to-guides/my-first-candy-machine-part1) but every "v3" mint-ui I've tinkered with using the latest sugar cli gives the error above. If i'm reading above correctly, is the latest Sugar CLI is making v2 candy machines that nothing currently supports?

**Comment by lorisleiva** [comment]
I know it's confusing but:
- Candy Machine V3 refer to the third iteration of the candy machine programs. It is composed of candy machine core and candy guard.
- When introducing programmable NFTs within "Candy Machine V3" we needed a way to distinguish the candy machine accounts that support PNFTs with the ones that don't. So we had to introduce a "version" number within the candy machine account of the candy machine core program.
- Therefore, when we are talking about version 1 and 2 in the context of the latest candy machine programs, we are talking about the account version and not the previous iteration of the program itself.

Again I'm sorry this has to be so confusing but does this make sense?

**Comment by Iceee1000** [comment]
Yes that explanation makes some sense. We've rolled back to using Sugar CLI 1.2.2 for work on our most recent collection, we'll revisit the new tool versions in the future...
